### PR TITLE
feat: kclrun controller code structure

### PR DESCRIPTION
**Pull request by ekkoful**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [x] N
- [ ] Y 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [x] Other

### CODE REVIEW

**Comment by ekkoful** [comment]
Generate the kubernetes gvk code and controller code structure. Also modify the project and go mod info to kcl-lang.io Please help double check. @Peefy

File: main.go [hunk 1]
```diff
@@ -67,7 +71,7 @@ func main() {
 		Port:                   9443,
 		HealthProbeBindAddress: probeAddr,
 		LeaderElection:         enableLeaderElection,
-		LeaderElectionID:       "f821e48b.my.domain",
+		LeaderElectionID:       "f821e48b.kcl-lang.io",
```
Comment (ekkoful) [inline main.go:74]: modify the domain to kcl-lang.io

File: go.mod [hunk 2]
```diff
@@ -1,8 +1,10 @@
-module github.com/KusionStack/kcl-operator
+module github.com/kcl-lang/kcl-operator
```
Comment (ekkoful) [inline go.mod:1]: update go mod module name cc: @Peefy

Comment (Peefy) [reply go.mod:1]: > update go mod module name cc: @Peefy

U can use kcl-lang.io/kcl-operator

**Pull request merged**